ext_localconf.php

```diff
@@ -18,25 +15,19 @@
         'quality' => 50,
         'os' => '',
         'exec' => '',
-        'className' => OAuth2LoginService::class
+        'className' => \Mfc\OAuth2\Services\OAuth2LoginService::class
     ]
 );
 
-if (isset($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['oauth2'])) {
-    $extensionConfig = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['oauth2']);
-} elseif (isset($GLOBALS['TYPO3_CONF_VARS']['EXTENSIONS']['oauth2'])) {
-    $extensionConfig = $GLOBALS['TYPO3_CONF_VARS']['EXTENSIONS']['oauth2'];
-} else {
-    $extensionConfig = [
-        'enableBackendLogin' => false,
-    ];
-}
+$enableBackendLogin = (bool)\TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(
+    \TYPO3\CMS\Core\Configuration\ExtensionConfiguration::class
+)->get('oauth2', 'enableBackendLogin');
 
-if ($extensionConfig['enableBackendLogin']) {
+if ($enableBackendLogin) {
     $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['backend']['loginProviders'][1529672977] = [
-        'provider' => OAuth2LoginProvider::class,
+        'provider' => \Mfc\OAuth2\LoginProvider\OAuth2LoginProvider::class,
         'sorting' => 25,
-        'icon-class' => 'fa-sign-in',
+        'iconIdentifier' => 'mfc-oauth2-login',
```

Comment:
TYPO3 core ships with an icon `actions-login`. I would suggest to use that instead of registering a dedicated icon.

https://typo3.github.io/TYPO3.Icons/icons/actions/actions-login.html